trie/stacktrie.go

```diff
@@ -64,7 +69,16 @@ func (t *StackTrie) Update(key, value []byte) error {
 	if len(value) == 0 {
 		return errors.New("trying to insert empty (deletion)")
 	}
-	k := t.TrieKey(key)
+	var k []byte
+	{ // Need to expand the 'key' into hex-form. We use the dedicated buf for that.
+		if cap(t.kBuf) < 2*len(key) { // realloc to ensure sufficient cap
+			t.kBuf = make([]byte, 2*len(key))
+		}
+		// resize to ensure correct size
+		t.kBuf = t.kBuf[:2*len(key)]
+		writeHexKey(t.kBuf, key)
+		k = t.kBuf
+	}
```

Comment:
Isn't `key` always 32 bytes here?

Reply:
No (according to test cases).

Reply:
Well, the parameter is a `[]byte`, not `[32]byte`. So the api allows for non-32 bytes. 

In practice, you're probably right. IMO the code should handle all sizes